client/my-sites/themes/index.web.js

```diff
@@ -5,14 +5,15 @@ import config from 'config';
 import userFactory from 'lib/user';
 import { makeLayout } from 'controller';
 import { makeNavigation, siteSelection, makeSites } from 'my-sites/controller';
-import { singleSite, multiSite, loggedOut, upload } from './controller';
-import { getSubjects } from './theme-filters';
-import validateFilters from './validate-filters';
+import { loggedIn, loggedOut, upload, fetchThemeFilters } from './controller';
+import { validateFilters, validateVertical } from './validate-filters';
 
 export default function( router ) {
 	const user = userFactory();
 	const isLoggedIn = !! user.get();
-	const verticals = getSubjects().join( '|' );
+	const siteId = '\\d+' + // numeric site id
+		'|' + // or
+		'[^\\\\/.]+\\.[^\\\\/]+'; // one-or-more non-slash-or-dot chars, then a dot, then one-or-more non-slashes
```

Comment:
Unfortunately, using a regex for the site id (and ordering the routes so we only match verticals after other routes fail) is what lets us accept anything for the vertical. If we don't enforce the site id it will match verticals, so a request `/themes/music` will match on [this route](url).